contracts/modules/v2/StakingPool.sol

```diff
@@ -747,15 +759,53 @@ contract StakingPool is IStakingPool, ERC721 {
   /* nft */
 
   function _beforeTokenTransfer(
-    address from,
+    address /*from*/,
     address /*to*/,
     uint256 tokenId
   ) internal view override {
-    require(
-      tokenId != 0 || nxm.isLockedForMV(from) < block.timestamp,
-      "StakingPool: Locked for voting in governance"
-    );
-    // todo: track owned zero-id nfts in TC
+    if(tokenId == 0) {
+      require(
+        nxm.isLockedForMV(manager()) < block.timestamp,
```

Comment:
Can you explain the change from `from` to `manager()`?

Reply:
The manager's lock is the one that matters. When the manager is locked, his manager NFT is non-transferable for 3 days. Other NFTs remain transferable since all withdrawals are halted so there's no double voting.

Reply:
We're inside the if condition. The `from` has to be the manager address so there's no point in fetching it from the storage. If the `from` is not the manager it will revert inside the ERC721 transfer function.

Reply:
What if it's an approved contract?